kernel/arch/dreamcast/hardware/maple/maple_init_shutdown.c

```diff
@@ -96,7 +80,8 @@ static void maple_hw_init(void) {
 
     /* Hook the necessary interrupts */
     maple_state.vbl_handle = vblank_handler_add(maple_vbl_irq_hnd, &maple_state);
-    asic_evt_set_handler(ASIC_EVT_MAPLE_DMA, maple_dma_irq_hnd, &maple_state);
+    asic_evt_request_threaded_handler(ASIC_EVT_MAPLE_DMA, maple_dma_irq_hnd,
```

Comment:
maple_dma_irq_hnd isn't safe to be run outside of an irq. It's not a "one line change" to set this into threading, even if it does 'just run'.

maple_dma_irq_hnd changes the frame queue, presuming it's inside an irq. Considering you weren't aware that most of this was happening in an irq context, please review the irq handler for concurrency implications beyond the most obvious one I pointed out.